src/python/pants/backend/python/targets/python_distribution.py

```diff
@@ -4,18 +4,15 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-from builtins import str
```

Comment:
why did you remove this? Is it unused now?

Reply:
Yes, because the `str()` call was only in the reimplementation of the `PythonTarget` constructor that was removed in favor of subclassing `PythonTarget` -- this line was causing the lint shard to fail. I was actually meaning to ask you about how to handle this in the future -- is there a linter for python 3 compat that would do this? I could probably write one from the tools in `contrib/python` if that would help to ensure the incorrect `str()` doesn't creep back in, I did something like that once (then removed it) when I got a bit overeager in a PR a while ago.

Reply:
Okay, sounds good.

I don't know of any linter that checks for use of the `future` library. PyLint has a mode for checking Py3 incompatibilities, but it's not checking for use of future, as I understand.

We also unfortunately can't start running `futurize` on every file committed, because it makes over-optimizations.

The ideal linter would check for any usages of `str, map, filter, bytes, zip, range, object` and other builtins, and make sure that a corresponding `from builtins import x` statement is used. That way we always are using Py3 syntax. That could be an easy win.

Reply:
Made into #6157!